src/UI/examples/Modal/Roundtrip/show_multi_step_modal.php

```diff
@@ -0,0 +1,45 @@
+<?php
+function show_multi_step_modal()
+{
+	global $DIC;
+	$f = $DIC->ui()->factory();
+	$r = $DIC->ui()->renderer();
+
+	$url = $_SERVER['REQUEST_URI'];
+
+	$page = $_GET["page"];
+	if ($page == "")
+	{
+		$modal = $f->modal()->roundtrip("Modal Title", $f->legacy("b"));
+		$asyncUrl = $url . '&page=login&replaceSignal=' . $modal->getReplaceContentSignal()->getId();
+		$modal = $modal->withAsyncRenderUrl($asyncUrl);
```

Comment:
This seems to be somehow duplicated (which is not the problem of this PR only, though). Do we really need this kind of functionality on the modal and on a rather similar signal? Is there a way to remove the version on the modal itself? The redundancy also shows in the js-Code. Could we get rid of duplicates there at least?

Reply:
Where in these lines do you identify duplication exactly?

Reply:
I see it in the functionality: There is a `withAsynRenderUrl` on the modal and a signal with `withAsyncRenderUrl` and they both seem to do almost the same. Only difference is, that the former is triggered "on modal load" while the latter is triggered by some other means. This also is reflected in the js-code: There are two locations, where some content is inserted/replaced, i.e. `showModal` and `replaceContentFromSignal`. Both locations differ in implementation, though. This seems odd at best and imo hints at a chance for some unification. Does this make any sense?